src/TuDa.CIMS.Api/Repositories/AssetItemRepository.cs

```diff
@@ -172,4 +181,135 @@ public async Task<IEnumerable<AssetItem>> SearchAsync(string nameOrCas)
+        {
+            return Error.NotFound(
+                "Assetitem.create",
+                $"Given RoomId {createModel.RoomId} was not found."
+            );
+        }
+
+        List<Hazard> hazards = [];
+        switch (createModel)
+        {
+            case CreateSolventDto createSolvent:
+                {
+                    foreach (var hazardId in createSolvent.Hazards)
+                    {
+                        var hazard = await _context.Hazards.SingleOrDefaultAsync(h => h.Id != hazardId);
+                        if (hazard is null)
+                        {
+                            return Error.NotFound(
+                                "Assetitem.create",
+                                $"Given HazardId {hazardId} was not found."
+                            );
+                        }
+                        hazards.Add(hazard);
+                    }
+
+                    break;
+                }
+            case CreateGasCylinderDto createGasCylinder:
+                {
+                    foreach (var hazardId in createGasCylinder.Hazards)
+                    {
+                         var hazard = await _context.Hazards.SingleOrDefaultAsync(h => h.Id != hazardId);
+                        if (hazard is null)
+                        {
+                            return Error.NotFound(
+                                "Assetitem.create",
+                                $"Given HazardId {hazardId} was not found."
+                            );
+                        }
+                        hazards.Add(hazard);
+                    }
+
+                    break;
+                }
+            case CreateChemicalDto createChemical:
+                {
+                    foreach (var hazardId in createChemical.Hazards)
+                    {
+                        var hazard = await _context.Hazards.SingleOrDefaultAsync(h => h.Id != hazardId);
+                        if (hazard is null)
+                        {
+                            return Error.NotFound(
+                                "Assetitem.create",
+                                $"Given HazardId {hazardId} was not found."
+                            );
+                        }
+                        hazards.Add(hazard);
+                    }
+
+                    break;
+                }
+        }
+
+        AssetItem newItem = createModel switch
+        {
+            CreateSolventDto solvent => new Solvent
+            {
+                ItemNumber = createModel.ItemNumber,
+                Name = createModel.Name,
+                Note = createModel.Note,
+                Price = createModel.Price,
+                Shop = createModel.Shop,
+                Room = room,
+                Cas = solvent.Cas,
+                Hazards = hazards,
+                PriceUnit = solvent.PriceUnit,
+                BindingSize = solvent.BindingSize,
+                Purity = solvent.Purity,
+            },
```

Comment:
🔧 Could maybe be simplified with Mapster.
```suggestion
            CreateSolventDto solvent => solvent.Adapt<Solvent>(),
```

Reply:
But Room and Hazards would need to be added manually